infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java

```diff
@@ -0,0 +1,376 @@
+            throw new IllegalArgumentException("The number of instances to create must be specified");
+        }
+
+        if (parameters[18] == null) {
+            throw new IllegalArgumentException("The number of nodes per instance to deploy must be specified");
+        }
+
+        if (parameters[19] == null) {
+            throw new IllegalArgumentException("The download node.jar command must be specified");
+        }
+
+        if (parameters[22] == null) {
+            parameters[22] = "";
+        }
+    }
+
+    @Override
+    public void acquireNode() {
+
+        connectorIaasController.waitForConnectorIaasToBeUP();
+
+        connectorIaasController.createAzureInfrastructure(getInfrastructureId(),
+                                                          clientId,
+                                                          secret,
+                                                          domain,
+                                                          subscriptionId,
+                                                          authenticationEndpoint,
+                                                          managementEndpoint,
+                                                          resourceManagerEndpoint,
+                                                          graphEndpoint,
+                                                          false);
+
+        String instanceTag = getInfrastructureId();
+        Set<String> instancesIds;
+        instancesIds = connectorIaasController.createAzureInstances(getInfrastructureId(),
+                                                                    instanceTag,
+                                                                    image,
+                                                                    numberOfInstances,
+                                                                    vmUsername,
+                                                                    vmPassword,
+                                                                    vmPublicKey,
+                                                                    vmSizeType,
+                                                                    resourceGroup,
+                                                                    region,
+                                                                    privateNetworkCIDR,
+                                                                    staticPublicIP);
+
+        logger.info("Instances ids created : " + instancesIds);
+
+        /*
+         * TODO: could be injected on boot just like OpenStack BUT scripts need to be customized
+         * with instanceTag/id
+         * and therefore we need to create the VMs one by one so the parallel instances creation
+         * feature from Azure
+         * will be lost.
+         */
+        for (String instanceId : instancesIds) {
+
+            String fullScript = "-c '" + this.downloadCommand + ";nohup " +
```

Comment:
The `generateDefaultStartNodeCommand()` method takes the current instance ID in parameter so the fullScript String variable is customized for each instance and cannot be move outside the loop.